src/mongocxx/test/spec/initial_dns_seedlist_discovery.cpp

```diff
@@ -0,0 +1,320 @@
+    switch (expected_option.type()) {
+        case bsoncxx::type::k_int32:
+            REQUIRE(expected_option.get_int32() == my_option.get_int32());
+            break;
+        case bsoncxx::type::k_bool:
+            REQUIRE(expected_option.get_bool() == my_option.get_bool());
+            break;
+        case bsoncxx::type::k_string:
+            REQUIRE(expected_option.get_string() == my_option.get_string());
+            break;
+        default:
+            std::string msg =
+                "option type not handled: " + bsoncxx::to_string(expected_option.type());
+            throw std::logic_error(msg);
+    }
+}
+
+static void compare_options(bsoncxx::document::view_or_value expected_options,
+                            bsoncxx::document::view_or_value my_options,
+                            bsoncxx::stdx::optional<bsoncxx::document::view> creds) {
+    auto my_options_view = my_options.view();
+    for (const auto& expected_option : expected_options.view()) {
+        auto key = std::string(expected_option.key());
+        std::transform(key.begin(), key.end(), key.begin(), [](int c) { return std::tolower(c); });
+        if (key == "ssl") {
+            key = std::string("tls");
+        }
+
+        bsoncxx::document::element my_value = my_options_view[key];
+        if (my_value) {
+            assert_elements_equal(expected_option, my_value);
+        } else if (creds && creds.value()[key]) {
+            assert_elements_equal(expected_option, creds.value()[key]);
+        } else {
+            FAIL("neither options nor credentials contains the required key: " + key);
+        }
+    }
+}
+
+static bool hosts_are_equal(bsoncxx::array::view expected_hosts,
+                            std::vector<std::string> actual,
+                            std::mutex& mtx) {
+    const std::lock_guard<std::mutex> lock(mtx);
+
+    std::vector<std::string> expected;
+
+    for (const auto& i : expected_hosts) {
+        expected.push_back(std::string(i.get_string().value));
+    }
+
+    std::sort(expected.begin(), expected.end());
+    std::sort(actual.begin(), actual.end());
+
+    return expected == actual;
+}
+
+static void validate_srv_max_hosts(mongocxx::client& client,
+                                   const InitialDNSSeedlistTest& test,
+                                   std::mutex& mtx,
+                                   std::vector<std::string>& new_hosts) {
+    using namespace mongocxx;
+    using bsoncxx::builder::basic::kvp;
+    using bsoncxx::builder::basic::make_document;
+
+    auto admin = client["admin"];
+    auto result = admin.run_command(make_document(kvp("ping", 1)));
+    REQUIRE(result.begin()->get_double() == 1.0);
+
+    auto uri = client.uri();
+    auto my_options = uri.options();
+    auto creds = uri.credentials();
+    compare_options(test.options, my_options, creds);
+
+    bool has_hosts_list = !test.hosts.empty();
+    if (test.num_hosts || has_hosts_list) {
```

Comment:
Tests specify "numHosts" instead of "hosts". "numHosts" is used when it is not possible to compare the hosts returned.

Here is one test:
https://github.com/mongodb/specifications/blob/31bbf60c4f3008d77fc5ca5287b165e5ea97a4a6/source/initial-dns-seedlist-discovery/tests/sharded/srvMaxHosts-less_than_srv_records.json#L4

Since no tests are asserting "numHosts", I suggest removing the "numHosts" logic. Do not parse "numHosts" in `InitialDNSSeedlistTest::parse` so an error will be thrown when tests are added with the "numHosts" assertion.